accounts-db/src/append_vec.rs

```diff
@@ -510,7 +512,7 @@ impl AppendVec {
     /// that falls on a 64-byte boundary.
     fn get_type<T>(&self, offset: usize) -> Option<(&T, usize)> {
         let (data, next) = self.get_slice(offset, mem::size_of::<T>())?;
-        let ptr: *const T = data.as_ptr() as *const T;
+        let ptr = data.as_ptr().cast();
         //UNSAFE: The cast is safe because the slice is aligned and fits into the memory
```

Comment:
I think that this comment is lying? The slice is &[u8] which has alignment=1.
What is making sure that data.as_ptr() is aligned to `mem::align_of::<T>()`?